tests/PHPStan/Rules/Arrays/NonexistentOffsetInArrayDimFetchRuleTest.php

```diff
@@ -942,6 +942,22 @@ public function testBug12593(): void
 		$this->analyse([__DIR__ . '/data/bug-12593.php'], []);
 	}
 
+	public function testBug12981(): void
+	{
+		$this->reportPossiblyNonexistentGeneralArrayOffset = true;
+
+		$this->analyse([__DIR__ . '/data/bug-12981.php'], [
+			[
+				'Offset array<int, int|string> might not exist on non-empty-array<bool|float|int|string>.',
```

Comment:
I feel like a good improvement would be to not have this error reported because it's already reported as a `Invalid array key type`.

https://phpstan.org/r/b4c81787-7c0b-46b3-b281-dd572ab6f8ac

But that's another issue